src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -79,7 +79,7 @@ public static AnalyzeResult<TItem> Analyze<TItem>(this GraphNode<TItem> root)
                 {
                     foreach (var sideNode in n.InnerNodes)
                     {
-                        if (sideNode != node && sideNode.Key.Name == node.Key.Name)
+                        if (sideNode != node && StringComparer.OrdinalIgnoreCase.Equals(sideNode.Key.Name, node.Key.Name))
```

Comment:
Key should be the library identity here. I wouldn't expect it to be null, but it can happen anywhere. The graph which you see here is very open in the resolver, and most of the code works by relying on the graph being sane.